packages/http/src/mocker/HttpMocker.ts

```diff
@@ -37,46 +37,71 @@ export class HttpMocker implements IMocker<IHttpOperation, IHttpRequest, IHttpCo
       throw new Error('Http request is not defined');
     }
 
-    // setting default values
-    const acceptMediaType = input.data.headers && caseless(input.data.headers).get('accept');
-    config = config || { mock: false };
-    const mockConfig: IHttpOperationConfig =
-      config.mock === false ? { dynamic: false } : Object.assign({}, config.mock);
+    let payloadGenerator: PayloadGenerator = generateStatic;
 
-    if (!mockConfig.mediaTypes && acceptMediaType) {
-      mockConfig.mediaTypes = acceptMediaType.split(',');
+    if (config && typeof config.mock !== 'boolean' && config.mock.dynamic) {
+      payloadGenerator = generate;
     }
 
-    // looking up proper example
-    let negotiationResult: IHttpNegotiationResult;
-    if (input.validations.input.length > 0) {
-      try {
-        negotiationResult = helpers.negotiateOptionsForInvalidRequest(resource.responses);
-      } catch (error) {
-        throw ProblemJsonError.fromTemplate(
-          UNPROCESSABLE_ENTITY,
-          `Your request body is not valid: ${JSON.stringify(input.validations.input)}`,
-        );
+    return withLogger(logger => {
+      // setting default values
+      const acceptMediaType = input.data.headers && caseless(input.data.headers).get('accept');
+      config = config || { mock: false };
+      const mockConfig: IHttpOperationConfig =
+        config.mock === false ? { dynamic: false } : Object.assign({}, config.mock);
+
+      if (!mockConfig.mediaTypes && acceptMediaType) {
+        logger.info(`Request contains an accept header: ${acceptMediaType}`);
+        mockConfig.mediaTypes = acceptMediaType.split(',');
       }
-    } else {
-      negotiationResult = helpers.negotiateOptionsForValidRequest(resource, mockConfig);
-    }
 
-    const mockedBody = computeBody(negotiationResult, payloadGenerator);
-    const mockedHeaders = computeMockedHeaders(negotiationResult.headers || [], payloadGenerator);
+      return mockConfig;
+    })
+      .chain(mockConfig => {
```

Comment:
can the function passed to `.chain` be moved out?

Reply:
Yes! Done in 34638c0 with the new `negotiateResponse` function!

Reply:
there even is `if/else`! I love it :+1: